src/bootstrap/test.rs

```diff
@@ -122,7 +122,17 @@ impl Step for Linkcheck {
 
     fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
         let builder = run.builder;
-        run.path("src/tools/linkchecker").default_condition(builder.config.docs)
+        let run = run.path("src/tools/linkchecker");
+
+        // if we have different hosts and targets, some things may be built for
+        // the host (e.g. rustc) and others for the target (e.g. std). The
+        // documentation built for each will contain broken links to
+        // docs built for the other platform (e.g. rustc linking to cargo)
+        if builder.hosts == builder.targets {
+            run.default_condition(builder.config.docs)
+        } else {
+            run.default_condition(false)
+        }
```

Comment:
I don't quite understand what you're saying here - I would expect our docs to not be missing parts when generated for a given host or target? Can you elaborate more on exactly what you're seeing (e.g., what broken link)?

Reply:
rustc docs are built into `build/x86_64-unknown-linux-gnu/doc`. All other docs are built to `build/riscv64gc-unknown-linux-gnu/doc/`, so when linkcheck looks at the rustc interdoc links it doesn't find the other documentation.

The missing links are 
```
rustc/lints/listing/warn-by-default.html:387: broken link - rustdoc/lints.html
rustc/lints/listing/warn-by-default.html:1037: broken link - rustdoc/lints.html
rustc/lints/listing/warn-by-default.html:1250: broken link - rustdoc/lints.html
rustc/lints/listing/warn-by-default.html:1518: broken link - rustdoc/lints.html
rustc/lints/listing/allowed-by-default.html:416: broken link - rustdoc/lints.html
rustc/lints/listing/allowed-by-default.html:568: broken link - rustdoc/lints.html
rustc/lints/listing/allowed-by-default.html:603: broken link - rustdoc/lints.html
rustc/lints/listing/allowed-by-default.html:709: broken link - rustdoc/lints.html
rustc/index.html:171: broken link - cargo/index.html
rustc/what-is-rustc.html:171: broken link - cargo/index.html
rustc/print.html:173: broken link - cargo/index.html
rustc/print.html:221: broken link - reference/conditional-compilation.html
rustc/print.html:250: broken link - reference/items/external-blocks.html
rustc/print.html:273: broken link - reference/linkage.html
rustc/print.html:276: broken link - reference/linkage.html
rustc/print.html:283: broken link - edition-guide/introduction.html
rustc/print.html:327: broken link - reference/conditional-compilation.html
rustc/print.html:414: broken link - reference/items/extern-crates.html
rustc/print.html:1051: broken link - rustdoc/lints.html
rustc/print.html:1203: broken link - rustdoc/lints.html
rustc/print.html:1238: broken link - rustdoc/lints.html
rustc/print.html:1344: broken link - rustdoc/lints.html
rustc/print.html:2077: broken link - rustdoc/lints.html
rustc/print.html:2727: broken link - rustdoc/lints.html
rustc/print.html:2940: broken link - rustdoc/lints.html
rustc/print.html:3208: broken link - rustdoc/lints.html
rustc/print.html:4926: broken link - reference/conditional-compilation.html
rustc/print.html:5176: broken link - reference/expressions/operator-expr.html
rustc/print.html:5306: broken link - reference/attributes/codegen.html
rustc/print.html:5310: broken link - reference/linkage.html
rustc/codegen-options/index.html:210: broken link - reference/conditional-compilation.html
rustc/codegen-options/index.html:460: broken link - reference/expressions/operator-expr.html
rustc/codegen-options/index.html:590: broken link - reference/attributes/codegen.html
rustc/codegen-options/index.html:594: broken link - reference/linkage.html
rustc/command-line-arguments.html:173: broken link - reference/conditional-compilation.html
rustc/command-line-arguments.html:202: broken link - reference/items/external-blocks.html
rustc/command-line-arguments.html:225: broken link - reference/linkage.html
rustc/command-line-arguments.html:228: broken link - reference/linkage.html
rustc/command-line-arguments.html:235: broken link - edition-guide/introduction.html
rustc/command-line-arguments.html:279: broken link - reference/conditional-compilation.html
rustc/command-line-arguments.html:366: broken link - reference/items/extern-crates.html
error-index.html:7: broken link - light1.51.0.css
error-index.html:8: broken link - rust.css
```